src/plugins/intel_npu/src/al/include/intel_npu/npu_mlir_runtime.hpp

```diff
@@ -0,0 +1,180 @@
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Runtime properties
+typedef struct _npu_mlir_runtime_properties_t {
+    uint32_t numOfSubGraphs;
+    uint32_t numOfGraphArgs;
+} npu_mlir_runtime_properties_t;
+
+typedef struct _npu_mlir_runtime_mem_ref_t {
+    const void* basePtr;
+    const void* data;
+    int64_t offset;
+    int64_t sizes[4];
+    int64_t strides[4];
+} npu_mlir_runtime_mem_ref_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Execute params
+typedef struct _npu_mlir_runtime_execute_params_t {
+    npu_mlir_runtime_mem_ref_t** pInputs;
+    uint32_t numOfInputs;
+    npu_mlir_runtime_mem_ref_t** pOutputs;
+    uint32_t numOfOutputs;
+    ze_context_handle_t ctx;
+    ze_device_handle_t device;
+    ze_graph_dditable_ext_t* graphDdiTableExt;
+    ze_command_list_handle_t* commandLists;
+    uint64_t numCommandLists;
+    ze_command_queue_handle_t commandQueue;
+    ze_fence_handle_t inferenceFence;
+    ze_event_handle_t event;
+} npu_mlir_runtime_execute_params_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Returns the API version
+NPU_MLIR_RUNTIME_APIEXPORT npu_mlir_runtime_result_t NPU_MLIR_RUNTIME_APICALL npuMLIRRuntimeGetAPIVersion(
+    npu_mlir_runtime_version_t* pVersion  ///< [out] pointer to version information
+);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Init MLIR runtime instance and return handle
+NPU_MLIR_RUNTIME_APIEXPORT npu_mlir_runtime_result_t NPU_MLIR_RUNTIME_APICALL npuMLIRRuntimeCreate(
+    const npu_mlir_runtime_blob_desc_t* desc,   ///< [in] pointer to graph descriptor
+    npu_mlir_runtime_handle_t* phRuntime,       ///< [out] pointer to handle of mlir runtime object created
+    npu_mlir_runtime_properties_t* pProperties  ///< [out] pointer to properties of the runtime
+);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Destroy MLIR runtime instance
+NPU_MLIR_RUNTIME_APIEXPORT npu_mlir_runtime_result_t NPU_MLIR_RUNTIME_APICALL npuMLIRRuntimeDestroy(
+    npu_mlir_runtime_handle_t hRuntime  ///< [in][release] handle of mlir runtime object to destroy
+);
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Get metadata from MLIR runtime instance
+NPU_MLIR_RUNTIME_APIEXPORT npu_mlir_runtime_result_t NPU_MLIR_RUNTIME_APICALL npuMLIRRuntimeGetMetadata(
+    npu_mlir_runtime_handle_t hRuntime,                          ///< [in] handle of mlir runtime object
+    uint32_t argIndex,                                           ///< [in] index of the argument
+    ze_graph_argument_properties_3_t* pGraphArgumentProperties,  ///< [out] query result for graph argument properties
+    ze_graph_argument_metadata_t* pGraphArgumentMetadata         ///< [out] query result for graph argument metadata
+);
```

Comment:
Let's add int64_t upperBound[5]. For no dynamic dimension, the bound will be zero. 
I will update get_metadata function to return upperBound.